### PR TITLE
Feature/cit 508/simplify logging

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

Simplification of logging
* The `mapper` and the `sm-agent` use the `tracing` crate for logging.
* The library crate can use the `log` crate: all the `error!`, `info!` emitted by such a library are catched up by `tracing_subscriber` of the `mapper` and the `sm-agent`.
* The `tedge` cli command directly writes to `stdout` and `stderr`.
* One doesn't use the `RUST_LOG` env variable to set the log level. Indeed, enabling the  `env-filter` feature of the tracing subscriber adds 600k on the stripped executable. It doesn't worth the value.
* Instead a `--debug` option has been added to the `mapper` and the `sm-agent`
    * Running without any option, the mapper and the agent use the `WARN` log level.
    * Running with `--debug`, the mapper and the agent use the `TRACE` log level.
* The spans of the main functions and loops of the agent and the mapper has been improved to provide a context to the log entries.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: sm/tedge_agent/src/main.rs [hunk 1]
```diff
@@ -16,10 +33,17 @@ async fn main() -> Result<(), anyhow::Error> {
...
+        tracing::Level::INFO
+    };
+
     tracing_subscriber::fmt()
         .with_timer(tracing_subscriber::fmt::time::ChronoUtc::with_format(
             "%Y-%m-%dT%H:%M:%S%.3f%:z".into(),
         ))
+        .with_max_level(log_level)
         .init();
 }
```
Comment (makr11st) [inline sm/tedge_agent/src/main.rs:49]: How about moving this function to `tedge_utils` (maybe `logging` module) crate such that we can have consistent logging and formatting (if we want to change date we only would have to change it one place).
Both `mapper` and `agent` use `tedge_utils` atm anyway so there won't be additional overhead with imports and in the future if we need to limit amount of `features` to be imported by other crates we can do something along the lines:
```rust
tedge_utils = { path = "../../common/tedge_utils", default-features = false, features = ["fs", "logging"] }
```
etc.

Comment (didier-wenzek) [reply sm/tedge_agent/src/main.rs:49]: That's a good point.

Comment (didier-wenzek) [reply sm/tedge_agent/src/main.rs]: See https://github.com/thin-edge/thin-edge.io/pull/427/commits/625744f5e2da4b3b96b889b7f1e2a19c3e7bfa72

**Review by makr11st — APPROVED** [review]
LGTM

**Review by PradeepKiruvale — COMMENTED** [review]
I did not get what was the criteria used for adding 
    #[instrument(skip(self), name = "")]
Added only some places and for some functions.

**Pull request merged**